### ISSUE TITLE
Performance with G-Helper is lower compared to when installing Armory Crate

### ISSUE THREAD

**Issue by jorismak** [seerge/g-helper#1353]
**Describe the bug**

I've done a complete Windows 11 clean reinstall on my Flow Z13, 2022 model (i9-12900h / 3050ti), GZ301ZE
I did not install Armory Crate (and even MyAsus), but I did install all the relevant driver packages from the Asus support page,
except the `Intel Converged Security and Management Engine Driver`, `ASUS Smart Display Control` and the `ASUS System Control Interface v3`.
Thinking 'lets first start clean, see what we get'.

I don't have the official charger plugged in, so I don't get any Turbo mode normally. What I notice from the previous (factory) install, is that 'silent' will limit the clockspeeds and power draw, and 'performance' will allow more, but also keep the clockspeed higher while idling. When setting Armory Crate to 'Windows' mode, it actually switches dynamically between lower clockspeeds and higher clockspeeds.

Anyway, now after this install, no Armory Crate, no MyAsus, but G-Helper for the first time.

I notice  that whatever performance mode I pick, I don't seem to get any different performance scores? Cinebench R23 single-run-multi-core almost always ends up between 9200 and 9600. While I could get higher on 'performance' mode on my previous install with Armory Crate installed. When running Cinebench, the clockspeeds also seem to just stop at 1.9ghz while CPU usage says +/- 72% which seems like a TDP- / power-limit?

Trying setting different power-limits in the advanced G-Helper profile screen doesn't seem to have _any_ effect at all.

I also seem to notice - but this is just a feeling - that the fan seems to stay lower-RPM ('not as noisy') as on my previous install, but when the fan does kick in, it kicks in hard and tends to go >6000 RPM and be very noticeable.  The Armory Crate install keps temps < 80 when not near 6000 RPM, while now the temps stay < 70C but the fan sits at 6000 or more a lot of the time.

When I go to the power settings in the Windows11 Settings, I only seem to have a single powerplan? It just says 'balanced', but I have no 'efficiency' or 'best performance' anymore there.

From the FAQ you seem to suggest that the fan-curve and power modes come from the firmware, but seeing this I highly doubt that this is the same.

I stopped G-Helper, rebooted, installed MyAsus and Armory Crate, let them check if there were drivers missing (they said no, but MyAsus was complaining that some services were stopped, with a repair button which I clicked), and rebooted again.

Behaviour is now back to normal so to speak. I can get Cinebench scores between the 11000 and 13000 again on 'performance' mode (with a regular 65watt PD charger, not the official 100W charger).

[log.zip](https://github.com/seerge/g-helper/files/12706483/log.zip)

Now, did I do something wrong in the install order? Did I need to install Armory Crate and MyAsus first, and then start G-Helper and let it disable all the Asus-services?
Or is there a chance the Z13 works just very differently (I believe you tested with AMD CPU-based systems, mostly).

I know other people actually changing and re-flashing the UEFI BIOS to alter power-limits and let it override the 'turbo' mode while using off-brand chargers. Which seem to suggest to me that it's the firmware in control of power modes like you said, but the behaviour on a clean Windows11 install without Asus software and only G-Helper is clearly different to Asus software installed and in use..

Armory Crate installed:
```
Item                       Version     
----                       -------     
Armoury Crate UWP App      5.6.10.0    
ROG Live Service           2.1.5.0     
Aura Service (Lighting Service)  3.07.17     
Armoury Crate service      5.6.10      
Aura Wallpaper Service     Not installed
Aura SDK                   3.04.32     
ASUS Aac_NBDT HAL          2.5.30.0    
```

MyAsus is at 3.1.26.0

**Comment by seerge** [comment]
@jorismak hello,

> When running Cinebench, the clockspeeds also seem to just stop at 1.9ghz while CPU usage says +/- 72% which seems like a TDP- / power-limit?

I think something is wrong with power plan (in windows), and your CPU simply are not allowed by system to boost / work harder :) 

I would suggest to check in HWinfo what kind of limits your CPU is hitting (and if it does hit them at all during benchmark) as most probably it's just windows not letting it boost.

Try to reset default Balanced power plan to defaults, and make sure that "power mode" (inside windows or under Fans+Power in G-Helper )

> Trying setting different power-limits in the advanced G-Helper profile screen doesn't seem to have any effect at all.
```
20/09/2023 23:28:05: PowerLimit A0 = 94 : OK
20/09/2023 23:28:05: PowerLimit A3 = 94 : OK
```

You have set them to 94W ? there is a hardcoded limits in BIOS for every model, and it won't let it go higher than that anyway. I can imagine it's way above that limits, so of course you won't see any difference. But you will see a difference, if you would set a limit to 30W for example

You can check your actual power consumption with HWInfo or similar app.

**Comment by seerge** [comment]
@jorismak PS after looking into your logs it looks you are mostly testing in Silent mode ? try Turbo or Balanced.

**Comment by jorismak** [comment]
I for sure tested with _mostly_ performance mode ('turbo').
Is the log file only for that last boot or something? That might explain why there is not much else in it.

**Comment by seerge** [comment]
@jorismak good to know, did resetting balanced power plan help?

**Comment by jorismak** [comment]
The whole part of the problem was that switching betewen modes (and running Cinebench in each mode) didn't show any difference in how clockspeeds and performance behaves. So I switched, for sure.

About TDP: On Armory Crate 'turbo' mode (only possible with official charger) there is actually a 115w powerlimit, according to Throttlestop. Also in reviews and reports about the device that it can draw (short time) over 100watts that the charger can deliver.

Anyway, I also tried with TDPs lower, but maybe I tried as low as in the 40s.

I tried G-Helper now again, (but this is with Armory Crate and everything installed) and it _clearly_ is different now. Although I also disabled E-cores in the BIOS at the moment, so it isn't an apples-to-apples comparison.

But setting it to 'Turbo' in G-Helper lets it boost to 4.2+ ghz all core (although again, all core is now 6 P-cores / 12 threads) for a short time before falling back to 3.16ghz and staying there for the run. Fans kick in of course but not as crazy loud as before.

Still get over 10.000 (sometimes) with just 6 P-cores.

'Balanced' keeps the fan at a very reasonable level and runs at 100% cpu usage but with cores at 2.8ghz, still a score of 8500 to 9000.

'Silent' keeps it dead silent, starts with a turbo of 100% at 3.1 ghz and then falls back to 90% cpu usage at 2.2 to 2.3ghz. Cinebench scores of over 7000. But again, dead silent in the fans, the whole run through.

Now, before installing Armory Crate, it would sit at 70% CPU usage at 1.9ghz no matter which mode is chosen in G-Helper.
The fans would also be very noise in all the modes.


These tests are done _now_ with 'power efficiency' mode in Win11 power settings:
![image](https://github.com/seerge/g-helper/assets/47725838/628d20bf-56d3-4198-99e4-12fa3f1bb3b9)

Before installing Armory Crate, I _only_ have 'Balanced' there, nothing more.

The older style 'power plans' also only showed Balanced before, but now they show 'Balanced' and 'Silent'.
In my previous install I had an 'ASUS Balanced' there as well?
![image](https://github.com/seerge/g-helper/assets/47725838/6a680c37-d94f-4ba0-a2e3-1cb692e5cbd9)


I kinda get back to my original assessment. It seems G-Helper is working very well, but only _after_ all the ASUS bloat is installed. Which makes perfect sense to me, but does tell me that some settings or drivers come with those tools that are needed. Because the 'bare BIOS firmware' modes don't seem to work as well.

(I also now see the power-limit is set to 66 for Turbo, 43 for Balanced and 125 (yes) for Silent.... before they were all set to 125w if I didn't touch the sliders).

I know that you probably can't do much without a similar device there, besides maybe saying somewhere "a single user reported better results with..." 😉 . I guess I have to en-bloat the device before I can de-bloat it :).
[log.zip](https://github.com/seerge/g-helper/files/12707239/log.zip)

**Comment by seerge** [comment]
@jorismak oke, good to know

I have re-read your original comment, and noticed that you wrote that you didn't install ``Asus System Control Interface``. That could be a reason as well (it's even [mentioned in requirements](https://github.com/seerge/g-helper#requirements-mandatory)). You might had another (older) version from windows update, or something. Probably it was reinstalled automatically along MyASUS / AC package :)

Most probably installing / uninstalling Asus stuff wasn't necessarily, but updating / reinstalling Control Interface (even via G-Helper's own Update section) could help :) if you want you can try that.

Just in case I have personally tried app on G14 2022, G14 2023, G16 2023 (with Intel CPU) + there is a lot of users with Z13 as well here ~ 8% https://github.com/seerge/g-helper/discussions/707 (from ~ 30-35K active users in total)

Do I understand that now everything works correctly for you and I can close this issue ?

**Comment by jorismak** [comment]
I stopped using G-Helper to install Armory Crate and related tools.

Now that I see that G-Helper actually does it's thing, I have it running
again and seems to be working fine!
I did the 'stop Asus services' thing and rebooted. I still get an Armory
Crate notification in Windows (!?!?) but G-Helper still reports all Asus
services stopped.

So, if it works like this: Awesome!

About 'Asus System Control Interface', I thought I read in the FAQ
somewhere that it should _not_ be installed. Maybe I just read that
backwards. I also still find it weird that a fresh Windows install came
with like, no 'power plans' or power modes whatsoever.

But I'm happy with how it's now, actually nice tool to have around (Because
it makes things way simpler / easier)!
